File: org-frontend/src/api/model/notebook.ts
// eslint-disable-next-line no-restricted-imports
import dayjs, { Dayjs } from "dayjs"

export interface Notebook {
  id: string
  user_id: string
  name: string
  description?: string
  created_at: Dayjs
  updated_at: Dayjs
  section_id?: string // Section assignment when note is in this notebook
}

export function fromJson(notebook: Notebook): Notebook {
  return {
    ...notebook,
    created_at: dayjs(notebook.created_at),
    updated_at: dayjs(notebook.updated_at),
  }
}

File: sigil-go/pkg/models/notebook.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Notebook struct {
	ID          uuid.UUID  `json:"id"                    db:"id"`
	UserID      uuid.UUID  `json:"user_id"               db:"user_id"`
	Name        string     `json:"name"                  db:"name"`
	Description string     `json:"description,omitempty" db:"description"`
	CreatedAt   time.Time  `json:"created_at"            db:"created_at"`
	UpdatedAt   time.Time  `json:"updated_at"            db:"updated_at"`
	SectionID   *uuid.UUID `json:"section_id,omitempty"  db:"section_id"` // Section assignment when note is in notebook
}
